src/pipeline.ts

```diff
@@ -317,30 +334,46 @@ export async function checkNewCandidateMatch(
 		);
 
 		if (!isAnyMatchedDecision(assessment.decision)) {
+			if (assessment.decision === Decision.SAME_INFO_HASH) {
+				decision = null;
+				break;
+			}
+			if (
+				assessment.decision === Decision.INFO_HASH_ALREADY_EXISTS &&
+				(!decision || !isAnyMatchedDecision(decision))
+			) {
+				decision = assessment.decision;
+			}
 			continue;
 		}
+		decision = assessment.decision;
 
-		result = await performAction(
-			assessment.metafile!,
-			assessment.decision,
-			searchee,
-			candidate.tracker,
-		);
+		actionResult = (
+			await performAction(
+				assessment.metafile!,
+				assessment.decision,
+				searchee,
+				candidate.tracker,
+			)
+		).actionResult;
```

Comment:
there's syntax sugar for this, but it's also a little weird—you can decide whether you like it better than this or not. 
```suggestion
		({ actionResult } = await performAction(
				assessment.metafile!,
				assessment.decision,
				searchee,
				candidate.tracker,
			))
```